dvc/config.py

```diff
@@ -250,7 +250,7 @@ class Config(object):  # pylint: disable=too-many-instance-attributes
     }
     COMPILED_SCHEMA = Schema(SCHEMA)
 
-    def __init__(self, dvc_dir=None, validate=True):
+    def __init__(self, dvc_dir=None, validate=True, verify_repo=False):
```

Comment:
Added a new flag because I didn't want to mess up with any previous logic dealing with having an empty `dvc_dir`.

Reply:
@mroutis This looks like a hack rather than a proper solution 🙂

Reply:
@efiop, could you explain? Not sure how to proceed with such observation 😅

Reply:
You are hacking your way through the bug instead of solving the bug. This `verify_repo` shouldn't be an option at all. Config should always raise if you try to modify in-repo config when there isn't one.